<p>My program has a very specific error that took me a while to track down - now I don't know how to fix it. My code is very long and in many files and I don't see much point in posting it here.</p>  <p>In the IDE, everything runs fine, in both Debug and Release (with the runtime library set to either /MTd or /MT, respectively, so I'm assuming all dependencies are included).</p>  <p>However, when I run the standalone, I get a crash. At first I thought it was a dependency problem but it doesn't seem so.</p>  <p>At some point in the code, I am accessing a vector via a method call: <code>t-&gt;GetList()</code>, where GetList is defined as <code>std::vector&lt;T*&gt; &amp; GetList()</code> and the method simply returns a member variable (simply defined as <code>std::vector&lt;T*&gt; field</code> in the class).</p>  <p>It turns out, if I print out the size of the list while running from the IDE, I get 0 (which is the correct answer in this case).</p>  <p>However, when doing the same thing running from standalone, I get the size as 467467353.</p>  <p>I tried changing the method declaration to <code>std::vector&lt;T*&gt; * GetList()</code> and doing <code>return &amp;field;</code> and updating the rest of the code but this didn't fix anything.</p>  <p>I then tried replacing the member variable <code>field</code> with a pointer, and in the constructor instantiating a new vector and deleting it in the destructor. Still no luck.</p>  <p>So I made a simple test case program that simply made a class with a vector field, and a method to return a reference to it. Then in <code>main</code> I would create an instance variable and getting the vector and printing the size. This worked both in VC++ and as a standalone - both returned zero.</p>  <p>This is driving me nuts and I can't figure out why I'm getting different behaviour. Why is the standalone running differently, and what should I be looking at to fix this?</p>  <p>Thanks</p>